src/main/java/com/redhat/cloud/notifications/routers/UserConfigService.java

```diff
@@ -89,15 +122,37 @@
     }
 
     @GET
-    @Path("/email-preference")
+    @Path("/notification-preference/{bundleName}/{applicationName}")
     @Operation(hidden = true)
-    public Uni<Response> getSettingsSchema(@Context SecurityContext sec) {
+    public Uni<UserConfigPreferences> getPreferences(
+            @Context SecurityContext sec,
+            @PathParam("bundleName") String bundleName,
+            @PathParam("applicationName") String applicationName) {
+        final RhIdPrincipal principal = (RhIdPrincipal) sec.getUserPrincipal();
+        final String account = principal.getAccount();
+        final String name = principal.getName();
+
+        Uni<EmailSubscription> uniDaily = emailSubscriptionResources.getEmailSubscription(account, name, bundleName, applicationName, EmailSubscriptionType.DAILY);
+        Uni<EmailSubscription> uniInstant = emailSubscriptionResources.getEmailSubscription(account, name, bundleName, applicationName, EmailSubscriptionType.INSTANT);
+
+        return Uni.combine().all().unis(uniDaily, uniInstant).combinedWith((daily, instant) -> {
+            UserConfigPreferences preferences = new UserConfigPreferences();
+            preferences.setDailyEmail(daily != null);
+            preferences.setInstantEmail(instant != null);
+            return preferences;
+        });
+    }
```

Comment:
This will execute the SQL queries concurrently because of `Uni.combine()`. Could you chain the two calls?

Something like this should work:
```
getEmailSubscription(DAILY)
.onItem().transform(daily -> /* create and return partial preferences */)
.onItem().transformToUni(preferences -> getEmailSubscription(INSTANT)
    .onItem().transform(instant -> /* add missing preferences field and return final object */)
)
```

Reply:
Sure.

Curious about this, is there any issue on executing both concurrently?

Reply:
With R2DBC it will work fine, but Hibernate will complain because the same `Session` is used concurrently.